client/components/SearchForm.jsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import _ from 'lodash';
+
+export default class SearchForm extends React.Component {
+  search(e) {
+    const searchQ = this.input.value
+    if (!searchQ) {
+        return this.props.onItemsChange([]);
+    }
+
+    $.get(`api/search/?q=${searchQ}`).done((data) => {
+      const items = JSON.parse(data).items;
+      this.props.onItemsChange(items);
```

Comment:
filter only the video items because this result contains playlists also. You should look for item.id.videoId, only videos have it.